### DISCUSSION TITLE
Sidebar

### DISCUSSION THREAD

**Discussion by d-oit** [gethinode/hinode#1120] [General]
Is there a reason that in the

layouts/_default/single.html

      {{ if $sidebar -}}

is queried instead of

    {{- $hasSidebar := .Site.Params.navigation.sidebar | default true -}}

 <div class="container-xxl flex-fill p-4 px-xxl-0">
        {{ if $hasSidebar -}}
            <div class="row row-cols-1 row-cols-{{ $breakpoint.current }}-2 row-cols-{{ $breakpoint.next }}-3">
                <div class="col col-{{ $breakpoint.next }}-2 d-none d-{{ $breakpoint.next }}-block sidebar-overflow sticky-top pt-5">
...

the current code always created a sidebar-overflow row

**Comment by markdumay** [comment]
You're right, it looks like sloppy coding. Replacing the final line in `layouts/_default/single/sidebar.html` to `{{ end -}}` (notice the additional minus sign, see https://pkg.go.dev/text/template#hdr-Text_and_spaces) would fix it. However, this remains somewhat fragile. I'll move the condition `{{- $hasSidebar := .Site.Params.navigation.sidebar | default true -}}` to `layouts/_default/single.html` as you suggested. I'll release a fix soon.

BTW, I'm not fully satisfied with the current base template and single/list templates. I plan to refactor and simplify these templates in the future.

**Reply by d-oit** [reply]
Thanks for the reference. 

Good to know that it is being planned.
Yes, there are a few problems with the current layout. Already adjust a few things adapted to my needs

- Not fully responsive mobile screens: top logo / hamburger menü is not visibile on screen if used long words in content:
e.g. BrokerageV1InstrumentsAsync(cancellationToken,instrumentId,with_attr,without_attr) 
- div with page content as printable / fullscreen area

**Reply by d-oit** [reply]
currently div "main-content" is strange at this point

![image](https://github.com/user-attachments/assets/c3d538d0-8618-4dd8-aae6-7d8145e1b4f6)
